cylc/flow/task_job_mgr.py

```diff
@@ -1000,16 +1000,17 @@ def _simulation_submit_task_jobs(self, itasks, workflow):
             itask.waiting_on_job_prep = False
             itask.submit_num += 1
             self._set_retry_timers(itask)
-            itask.platform = {'name': 'SIMULATION'}
+            itask.platform['name'] = 'SIMULATION'
             itask.summary['job_runner_name'] = 'SIMULATION'
             itask.summary[self.KEY_EXECUTE_TIME_LIMIT] = (
-                itask.tdef.rtconfig['job']['simulated run length']
+                itask.tdef.rtconfig['simulation']['simulated run length']
```

Comment:
I'm not sure this was wrong, because if it was then we would have got a `KerError` whenever this code was run, so `[job][simulated run length]` must have existed right?

Reply:
Simulated run length isn't in the config spec. AFAICT Cylc just uses it as a convenient place to dump this value. We could make it part of an object on the scheduler if you want? Like `sim_mode_manager`

Reply:
My point is that this appears to have been working before (otherwise we would have got traceback right?) so I'm not sure why it has changed here?

Reply:
> Did you change the output of the config stuff to move this from job to simulation?

No!